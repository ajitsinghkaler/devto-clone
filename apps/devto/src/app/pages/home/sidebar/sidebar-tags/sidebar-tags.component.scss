.tags-title {
  font-size: 16px;
  font-weight: 700;
  color: #202428;
}

header {
  padding: 0.5rem;
}

.followed-tags {
  height: 42vh;
  overflow-y: auto;
  a {
    display: block;
    padding: 0.5rem;
    color: #202428;
    border-radius: 5px;
    width: 100%;
    &:hover,
    &:active {
      background-color: rgba(8, 9, 10, 0.05);
      color: #323ebe;
    }
  }
}
