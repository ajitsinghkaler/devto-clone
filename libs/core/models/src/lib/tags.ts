export interface Tag {
  id: number;
  name: string;
  bg_color_hex: string;
  text_color_hex: string;
}
