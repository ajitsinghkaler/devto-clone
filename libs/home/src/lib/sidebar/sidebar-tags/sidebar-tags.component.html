<nav aria-label="Secondary sidebar nav">
  <header class="flex align-center justify-between">
    <h3 class="tags-title">My Tags</h3>
    <a
      href="/dashboard/following_tags"
      aria-label="Customize tag priority"
      title="Customize tag priority"
    >
      <svg
        xmlns="http://www.w3.org/2000/svg"
        width="24"
        height="24"
        viewBox="0 0 24 24"
        fill="#64707d"
      >
        <path
          d="M12 1l9.5 5.5v11L12 23l-9.5-5.5v-11L12 1zm0 2.311L4.5 7.653v8.694l7.5 4.342 7.5-4.342V7.653L12 3.311zM12 16a4 4 0 110-8 4 4 0 010 8zm0-2a2 2 0 100-4 2 2 0 000 4z"
        ></path>
      </svg>
    </a>
  </header>
  <div
    *rxLet="tags$; let tags; let err = $error"
    class="followed-tags"
    style="max-height: 42vh"
  >
    <div *ngFor="let tag of tags">
      <a title="{{ tag.name }} tag" href="/t/{{ tag.name }}">#{{ tag.name }}</a>
    </div>
  </div>
  <ng-template #suspense>loading</ng-template>
</nav>
