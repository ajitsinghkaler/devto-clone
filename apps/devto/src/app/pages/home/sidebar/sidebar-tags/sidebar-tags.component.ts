import { Component, OnInit } from '@angular/core';
import { TagsStore } from '../services/tags.store';

@Component({
  selector: 'app-sidebar-tags',
  templateUrl: './sidebar-tags.component.html',
  styleUrls: ['./sidebar-tags.component.scss'],
})
export class SidebarTagsComponent {
  tags$ = this.tagsStore.tags$;
  constructor(private tagsStore: TagsStore) {}
}
