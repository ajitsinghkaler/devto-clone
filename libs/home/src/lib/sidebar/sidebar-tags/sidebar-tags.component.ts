import { Component, OnInit } from '@angular/core';
import { TagsStore } from '../services/tags.store';

@Component({
  selector: 'app-sidebar-tags',
  template: `<nav aria-label="Secondary sidebar nav">
    <header class="flex align-center justify-between">
      <h3 class="tags-title">My Tags</h3>
      <a
        href="/dashboard/following_tags"
        aria-label="Customize tag priority"
        title="Customize tag priority"
      >
        <svg
          xmlns="http://www.w3.org/2000/svg"
          width="24"
          height="24"
          viewBox="0 0 24 24"
          fill="#64707d"
        >
          <path
            d="M12 1l9.5 5.5v11L12 23l-9.5-5.5v-11L12 1zm0 2.311L4.5 7.653v8.694l7.5 4.342 7.5-4.342V7.653L12 3.311zM12 16a4 4 0 110-8 4 4 0 010 8zm0-2a2 2 0 100-4 2 2 0 000 4z"
          ></path>
        </svg>
      </a>
    </header>
    <div
      *rxLet="tags$; let tags; let err = $error"
      class="followed-tags"
      style="max-height: 42vh"
    >
      <div *ngFor="let tag of tags">
        <a title="{{ tag.name }} tag" href="/t/{{ tag.name }}"
          >#{{ tag.name }}</a
        >
      </div>
    </div>
    <ng-template #suspense>loading</ng-template>
  </nav> `,
  styles: [
    `
      .tags-title {
        font-size: 16px;
        font-weight: 700;
        color: #202428;
      }

      header {
        padding: 0.5rem;
      }

      .followed-tags {
        height: 42vh;
        overflow-y: auto;
        a {
          display: block;
          padding: 0.5rem;
          color: #202428;
          border-radius: 5px;
          width: 100%;
          &:hover,
          &:active {
            background-color: rgba(8, 9, 10, 0.05);
            color: #323ebe;
          }
        }
      }
    `,
  ],
})
export class SidebarTagsComponent {
  tags$ = this.tagsStore.tags$;
  constructor(private tagsStore: TagsStore) {}
}
