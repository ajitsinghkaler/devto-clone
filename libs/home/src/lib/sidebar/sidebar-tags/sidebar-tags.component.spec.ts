import { ComponentFixture, TestBed } from '@angular/core/testing';
import { LetModule } from '@rx-angular/template';
import { of } from 'rxjs';
import { TagsStore } from '../services/tags.store';

import { SidebarTagsComponent } from './sidebar-tags.component';

describe('SidebarTagsComponent', () => {
  let component: SidebarTagsComponent;
  let fixture: ComponentFixture<SidebarTagsComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [LetModule],
      declarations: [SidebarTagsComponent],
      providers: [{ provide: TagsStore, useValue: mockTagsStore }],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(SidebarTagsComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render', () => {
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});

const mockTagsStore = {
  tags: of([
    {
      id: 6,
      name: 'javascript',
      bg_color_hex: '#F7DF1E',
      text_color_hex: '#000000',
    },
    {
      id: 8,
      name: 'webdev',
      bg_color_hex: '#562765',
      text_color_hex: '#ffffff',
    },
  ]),
};
