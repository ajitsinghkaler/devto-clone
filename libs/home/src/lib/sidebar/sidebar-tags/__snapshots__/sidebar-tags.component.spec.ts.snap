// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarTagsComponent should render 1`] = `
<div
  id="root1"
>
  <nav
    aria-label="Secondary sidebar nav"
  >
    <header
      class="flex align-center justify-between"
    >
      <h3
        class="tags-title"
      >
        My Tags
      </h3>
      <a
        aria-label="Customize tag priority"
        href="/dashboard/following_tags"
        title="Customize tag priority"
      >
        <svg
          fill="#64707d"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 1l9.5 5.5v11L12 23l-9.5-5.5v-11L12 1zm0 2.311L4.5 7.653v8.694l7.5 4.342 7.5-4.342V7.653L12 3.311zM12 16a4 4 0 110-8 4 4 0 010 8zm0-2a2 2 0 100-4 2 2 0 000 4z"
          />
        </svg>
      </a>
    </header>
    
    
  </nav>
</div>
`;
